group.c

```diff
@@ -58,14 +58,34 @@ static void group_remove(struct Group *g)
   FREE(&g);
 }
 
-int mutt_group_context_clear(struct GroupContext **ctx)
+/*　Please double check */
+/* And i will put original code into here in order to make convention for your review
+ * */
+
+/* original code*/
+/*
+ *
+ int mutt_group_context_clear(struct GroupContext **ctx)
+ {
+   struct GroupContext *t = NULL;
+   for (; ctx && *ctx; (*ctx) = t)
+   {
+     group_remove((*ctx)->g);
+     t = (*ctx)->next;
+     FREE(ctx);
+     }
+     return 0;
+  }
+ * */
+
+int mutt_group_context_clear(struct GroupContextHead *head)
 {
-  struct GroupContext *t = NULL;
-  for (; ctx && *ctx; (*ctx) = t)
+  struct GroupContext *np = NULL, *tmp = NULL;
+  STAILQ_FOREACH_SAFE(np, head, entries, tmp)
   {
-    group_remove((*ctx)->g);
-    t = (*ctx)->next;
-    FREE(ctx);
+    group_remove((np)->g);
+    STAILQ_REMOVE(head, np, GroupContext, entries);
+    FREE(&(np));
   }
```

Comment:
Please use the pattern used in `mutt_list_free` in `mutt/list.c` for a quicker removal of all elements in the list.

Please remove the parentheses around single tokens, as in `(np)->g` --> `np->g` and `FREE(&(np))` --> `FREE(&np)`.